prow/logrusutil/logrusutil.go

```diff
@@ -99,36 +98,26 @@ func (f CensoringFormatter) Format(entry *logrus.Entry) ([]byte, error) {
 	return f.censor(raw), nil
 }
 
-const censored = "CENSORED"
-
-var (
-	censoredBytes = []byte(censored)
-	standardLog   = logrus.NewEntry(logrus.New())
-)
-
 // Censor replaces sensitive parts of the content with a placeholder.
 func (f CensoringFormatter) censor(content []byte) []byte {
-	for _, secret := range f.getSecrets().List() {
-		trimmedSecret := strings.TrimSpace(secret)
-		if trimmedSecret != secret {
-			standardLog.Warning("Secret is not trimmed")
-			secret = trimmedSecret
-		}
-		if secret == "" {
-			standardLog.Warning("Secret is an empty string, ignoring")
-			continue
-		}
-		content = bytes.ReplaceAll(content, []byte(secret), censoredBytes)
-	}
+	f.censorer.Censor(content)
 	return content
 }
 
 // NewCensoringFormatter generates a `CensoringFormatter` with
 // a formatter as delegate and a set of strings to censor
 func NewCensoringFormatter(f logrus.Formatter, getSecrets func() sets.String) CensoringFormatter {
+	censorer := secretutil.NewCensorer()
+	censorer.Refresh(getSecrets().List()...)
```

Comment:
Technically speaking this will not refresh the list of secrets, but the previous implementation would. A quick survey of the users of this shows it's people shoving static lists in there ...

```go
logrus.SetFormatter(logrusutil.NewCensoringFormatter(logrus.StandardLogger().Formatter, func() sets.String {
	return sets.NewString(oauthToken)
}))
```